sdk/src/types/block/address/bech32.rs

```diff
@@ -172,11 +176,15 @@ impl Bech32Address {
 
 impl core::fmt::Display for Bech32Address {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-        write!(
-            f,
-            "{}",
-            bech32::encode::<bech32::Bech32>(self.hrp.0, &self.inner.pack_to_vec(),).unwrap()
-        )
+        let bytes = if self.inner.is_multi() {
+            let mut bytes = vec![MultiAddress::KIND];
+            bytes.extend(Blake2b256::digest(self.inner.pack_to_vec()));
```

Comment:
I don't really like that extend, any idea ?

Reply:
you could have fun with rshift 🤷‍♀️

Reply:
although I think the best solution may be a chain and collect since the packed vec will need to re-allocate anyway most likely

Reply:
Wouldn't that be the same amount of allocations?

Reply:
maybe. not sure if the ```let mut bytes = vec![MultiAddress::KIND]``` will need to reallocate because you didn't specify the capacity.

Reply:
Pretty sure extend will reallocate yes. You're saying that wouldn't be necessary with collect because of size hints ?

Reply:
I believe so yes. But that may actually be true with extend as well, the compiler is a mystery sometimes

Reply:
I changed to chain, I prefer it anyway